packages/web-api-scan-job-manager/src/worker/worker.ts

```diff
@@ -0,0 +1,114 @@
+
+    private jobId: string;
+    private jobManagerConfig: JobManagerConfig;
+    private restartAfterTime: Date;
+
+    public constructor(
+        @inject(Batch) private readonly batch: Batch,
+        @inject(Queue) private readonly queue: Queue,
+        @inject(PoolLoadGenerator) private readonly poolLoadGenerator: PoolLoadGenerator,
+        @inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration,
+        @inject(Logger) private readonly logger: Logger,
+    ) {}
+
+    public async run(): Promise<void> {
+        await this.init();
+
+        // tslint:disable-next-line: no-constant-condition
+        while (true) {
+            const poolMetricsInfo = await this.batch.getPoolMetricsInfo();
+            const tasksIncrementCount = await this.poolLoadGenerator.getTasksIncrementCount(poolMetricsInfo);
+
+            let tasksQueuedCount = 0;
+            if (tasksIncrementCount > 0) {
+                const scanMessages = await this.getMessages(tasksIncrementCount);
+                if (scanMessages.length === 0) {
+                    this.logger.logInfo(`The storage queue '${this.queue.scanQueue}' has no message to process.`);
+                    if (poolMetricsInfo.load.activeTasks === 0) {
+                        this.logger.logInfo(`Exiting the ${this.jobId} job since there are no active tasks.`);
+                        break;
+                    }
+                }
+
+                if (scanMessages.length > 0) {
+                    tasksQueuedCount = await this.addTasksToJob(scanMessages);
+                }
+            }
+
+            this.poolLoadGenerator.setLastTasksIncrementCount(tasksQueuedCount);
+
+            this.logger.logInfo('Pool load statistics', {
+                activeTasks: poolMetricsInfo.load.activeTasks.toString(),
+                runningTasks: poolMetricsInfo.load.runningTasks.toString(),
+                requestedTasksToAddPerInterval: tasksIncrementCount.toString(),
+                tasksAddedPerInterval: tasksQueuedCount.toString(),
+                samplingIntervalInSeconds: this.poolLoadGenerator.samplingIntervalInSeconds.toString(),
+                processingSpeedTasksPerMinute: this.poolLoadGenerator.processingSpeedPerMinute.toString(),
+                activeToRunningTasksRatio: this.poolLoadGenerator.activeToRunningTasksRatio.toString(),
+            });
+
+            if (this.runOnce) {
+                break;
+            }
+
+            if (moment().toDate() > this.restartAfterTime) {
+                this.logger.logInfo(`Performing scheduled termination after ${this.jobManagerConfig.maxWallClockTimeInHours} hours.`);
+                break;
+            }
+
+            await System.wait(this.jobManagerConfig.addTasksIntervalInSeconds * 1000);
+        }
+    }
+
+    private async getMessages(messagesCount: number): Promise<Message[]> {
+        const messages: Message[] = [];
+        do {
+            const batch = await this.queue.getMessages();
```

Comment:
can we rename this to queuedMessages / deQueuedMessages or something similar to be clearer about the value in this const?

Reply:
I think the name makes sense cause we only read the messages, and later we remove it, so it's not dequeued yet

Reply:
Honestly, `batch` does not tell me anything about what the variable is holding. What is it a batch of ? We don't have the concept of batch at all. Some other options you might want to consider - `fetchedMessages` `scanRequestMessages` `messagesToScan` `scanRequests`. All of these tell you something about what the variable is holding.